src/test/groovy/unit/org/pih/warehouse/inventory/CycleCountProductAvailabilityServiceSpec.groovy

```diff
@@ -361,27 +353,28 @@ class CycleCountProductAvailabilityServiceSpec extends Specification implements
         Product product = new Product()
         InventoryItem inventoryItem = new InventoryItem(product: product, lotNumber: 'lotNumber')
         Location binLocation = new Location(name: 'binLocation')
+        CycleCountItem countItem = new CycleCountItem(
+                inventoryItem: inventoryItem,
+                location: binLocation,
+                product: product,
+                countIndex: 0,  // count
+                quantityOnHand: 20,
+                custom: false,
+        )
+        countItem.id = '0'
```

Comment:
it's really frustrating that I had to do this but id still ends up as null if we put it in the constructor. It's some grails DataTest weirdness...
```
CycleCountItem countItem = new CycleCountItem(id: '1')
assert countItem.id == null
```

Reply:
AFAIR, in Grails, you cannot set the ID unless you use "assigned" as the generator.

Reply:
Actually I always thought it was due to the "generator", but it might just be that the id property is not bindable. https://stackoverflow.com/a/43546046/136597. 

I actually would like to be able to bind the ID for testing purposes so maybe we could look into making it bindable for all domains using the following in application.groovy. 
```
grails.gorm.default.mapping = {
    '*'(cascadeValidate: 'none')
    id generator: 'uuid', bindable: true
}
```

Reply:
oh interesting. Maybe best to do this only in tests at first (surely there's a way to define a test only config file) since we do a lot of param binding to domains:
```
new Domain(params)
```
which comes directly from the API request body so there might be some weird behaviour if the client is passing id in the request body and we suddenly start setting/saving it on the domain.